ads/insticator.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function insticator(global, data) {
+  // validate passed data attributes
+  validateData(data, ['siteId', 'embedId']);
+
+  // create insticator markup
+  global.document.getElementById('c').appendChild(createTemplate(data.embedId));
+
+  // create ads and embed
+  createAdsAndEmbed(data.siteId, data.embedId);
+
+  // envoke AMP library
+  loadScript(global);
+}
+
+// ------- HELPER VARIABLES ------- //
+// reusable URL references to ads, embed and the library
+// Don't forget to preconnect and prefetch as it's described in the docs
+// https://github.com/ampproject/amphtml/blob/master/ads/README.md
+const url = {content: '//d3lcz8vpax4lo2.cloudfront.net'};
+
+/**
+ * Create HTML template to be populated later
+ * @param {string} embedId The Unique Identifier of this particular Embed
+ * @return {element} HTML template
+ */
+function createTemplate(embedId) {
+  const template = document.createElement('template');
+  template.innerHTML = `
```

Comment:
you will need to mark this with `template./*OK*/.innnerHTML`

Reply:
When I add this I get an error in the console ...

`SyntaxError: /Users/troyriso/Desktop/workspace/amphtml/ads/insticator.js: Unexpected token (50:17)

  48 | function createTemplate(embedId) {
  49 |   const template = document.createElement('template');
> 50 |   template./*OK*/.innnerHTML = `
     |                  ^
  51 |     <div id="insticator-container">
  52 |       <div id="div-insticator-ad-1"></div>
  53 |       <div id="insticator-embed" embed-id="${embedId}"></div> while parsing file: /Users/troyriso/Desktop/workspace/amphtml/ads/insticator.js
[14:12:15] Error while compiling integration.js`

Reply:
ah sorry typo. There shouldn't be a second `.` 
`template./*OK*/innnerHTML`

Reply:
No worries. This one doesn't throw a console error, but now the embed doesn't load.

Reply:
Hmm, got it to work. Should be updated now.